<launch>
  <!-- Select your Reachy configuration -->
  <arg name="head" default="false" />
  <arg name="left_arm" default="true" />
  <arg name="right_arm" default="true" />

  <!-- specify the planning pipeline -->
  <arg name="pipeline" default="ompl" />

  <!-- By default, we do not start a database (it can be large) -->
  <arg name="db" default="false" />
  <!-- Allow user to specify database location -->
  <arg name="db_path" default="$(find reachy_moveit_config)/default_warehouse_mongo_db" />

  <!-- By default, we are not in debug mode -->
  <arg name="debug" default="false" />

  <!--
  By default, hide joint_state_publisher's GUI
  MoveIt's "demo" mode replaces the real robot driver with the joint_state_publisher.
  The latter one maintains and publishes the current joint configuration of the simulated robot.
  It also provides a GUI to move the simulated robot around "manually".
  This corresponds to moving around the real robot without the use of MoveIt.
  -->
  <arg name="use_gui" default="false" />
  <arg name="use_rviz" default="true" />

  <node name="interpolator" pkg="colab_reachy_control" type="interpolator" respawn="true" output="screen">
    <param name="device" value="/dev/serial/by-id/usb-Xevelabs_USB2AX_74031303437351011190-if00" />
  </node>

  <node name="action_server" pkg="colab_reachy_control" type="action_server.py" respawn="true" output="screen">
    <param name="rate" value="30" />

    <!-- Amount of time (s) permitted to be late achieving goal -->
    <param name="goal_time_tolerance" value="5.0" />
  </node>

  <!-- <node name="move_to_goal" pkg="colab_reachy_control" type="move_to_goal" respawn="true" output="screen" /> -->

  <!-- Aggregate joint states from components -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="use_gui" value="$(arg use_gui)"/>
    <rosparam param="source_list">[left_arm_controller/joint_states, right_arm_controller/joint_states]</rosparam>
  </node>

  <!-- Given the published joint states, publish tf for the robot links -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />

  <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
  <include file="$(find reachy_moveit_config)/launch/move_group.launch">
    <arg name="head" value="$(arg head)" />
    <arg name="left_arm" value="$(arg left_arm)" />
    <arg name="right_arm" value="$(arg right_arm)" />
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="false"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="pipeline" value="$(arg pipeline)"/>
  </include>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <include file="$(find reachy_moveit_config)/launch/moveit_rviz.launch" if="$(arg use_rviz)">
    <arg name="rviz_config" value="$(find reachy_moveit_config)/launch/moveit.rviz"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>
</launch>
